File: python/roman_numerals_encoder/to_roman_numerals.py
def solution(n):
  roman = {1000: 'M', 900: 'CM', 500: 'D', 400: 'CD', 100: 'C', 90: 'XC', 50: 'L', 40: 'XL', 10: 'X', 9: 'IX', 5: 'V', 4: 'IV', 1: 'I'}

  if n in roman:
    return roman[n]

  res = ''
  for k, v in roman.items():
    x, y = divmod(n, k)
    res += v * x
    n = y

  return res

File: javascript/roman_numerals_encoder/to_roman_numerals.js
function solution(number) {
  const roman = { M: 1000, CM: 900, D: 500, CD: 400, C: 100, XC: 90, L: 50, XL: 40, X: 10, IX: 9, V: 5, IV: 4, I: 1 }
  let result = ''

  for (key in roman) {
    result += key.repeat(Math.floor(number / roman[key]))
    number %= roman[key]
  }

  return result
}
module.exports = solution;
